### PR TITLE
Simplify symlink creation, update dependencies

### PR DESCRIPTION
**Pull request by MarkEWaite**
## Update dependencies

Use the current parent pom and other dependencies.  Update the spotbugs exclusions for new warnings reported with the most recent spotbugs.  Create symbolic links and hard links with Java NIO methods instead of making calls to the operating system.  The change to Java NIO calls avoids a spotbugs warning related to ProcessBuilder command injection.

Exclude okhttp and annotation-api from fastjson because newer versions are being provided by other dependencies.

Update owasp-java-html-sanitizer to 20211018.2 to resolve a security warning.

- Use fastjson 1.2.79, not 1.2.73
- Use okhttp-urlconnection 4.9.3, not 4.8.0
- Use commons-io 2.11.0, not 2.7
- Use commons-lang3 3.12.0, not 3.11
- Use commons-codec 1.15, not 1.14
- Use support-log-formatter 1.1, not 1.0
- Use version-number 1.9, not 1.7
- Use ant 1.10.12, not 1.10.11
- Use owasp-java-html-sanitizer 20211018.2, not 20200713.1
- Use spotbugs-annotations 4.5.3, not 4.1.1
- Use jsoup 1.14.3, not 1.14.2
- Use kotlin-stdlib-common 1.6.10, not 1.3.72
- Use mockito-core 4.3.1, not 3.4.6
- Use junit 4.13.2, not 4.13.1
- Use mockwebserver 4.9.3, not 4.8.0
- Use parent pom 1.70, not 1.54
- Use spotbugs 4.5.3.0, not 4.2.0
- Use surefire 3.0.0-M5, not 2.18.1
- Exclude okhttp and annotation-api from fastjson
- Exclude several spotbugs warnings
- Allow catch of NPE in getMetaData()
- Exclude command injection vulnerability
- Replace ProcessBuilder with Java 8 file calls
- Exclude path traversal spotbugs warnings

### CODE REVIEW

**Review by basil — APPROVED** [review]

**Comment by daniel-beck** [comment]
Thanks!

File: src/main/java/io/jenkins/update_center/DirectoryTreeBuilder.java [hunk 1]
```diff
@@ -157,21 +156,15 @@ protected void stage(MavenArtifact a, File dst) throws IOException {
...
         try {
-            if (p.waitFor() != 0) {
-                throw new IOException("'ln -f " + src.getAbsolutePath() + " " + dst.getAbsolutePath() +
-                        "' failed with code " + p.exitValue() + "\nError: " + IOUtils.toString(p.getErrorStream()) + "\nOutput: " + IOUtils.toString(p.getInputStream()));
-            } else {
-                LOGGER.log(Level.INFO, "Created new download file " + dst + " from " + src);
-            }
-        } catch (InterruptedException ex) {
-            LOGGER.log(Level.WARNING, "Interrupted creating " + dst + " from " + src, ex);
+            Files.createSymbolicLink(newLink, existingFile);
```
Comment (daniel-beck) [inline src/main/java/io/jenkins/update_center/DirectoryTreeBuilder.java:165]: This is wrong: `ln` first takes the destination file, then the link file argument. This switches the order around.

File: src/main/java/io/jenkins/update_center/DirectoryTreeBuilder.java [hunk 2]
```diff
@@ -157,21 +156,15 @@ protected void stage(MavenArtifact a, File dst) throws IOException {
             throw new IOException("Failed to create " + parentFile);
         }
 
-        ProcessBuilder pb = new ProcessBuilder();
         if (System.getProperty("os.name").toLowerCase(Locale.US).contains("windows")) {
             return;
         }
-        pb.command("ln", "-f", src.getAbsolutePath(), dst.getAbsolutePath());
```
Comment (daniel-beck) [inline src/main/java/io/jenkins/update_center/DirectoryTreeBuilder.java:164]: This creates a hard link, not a symlink, so this change is wrong.

Comment (daniel-beck) [reply src/main/java/io/jenkins/update_center/DirectoryTreeBuilder.java:164]: Caused https://github.com/jenkins-infra/helpdesk/issues/3095

**Pull request merged**